app/Http/Controllers/Backend/Transport/StationController.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace App\Http\Controllers\Backend\Transport;
+
+use App\Exceptions\HafasException;
+use App\Http\Controllers\Controller;
+use App\Http\Controllers\HafasController;
+use App\Models\TrainStation;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+
+abstract class StationController extends Controller
+{
+
+    /**
+     * @throws HafasException
+     * @throws ModelNotFoundException
+     */
+    public static function lookupStation(string|int $query): TrainStation {
+        //Lookup by station id
+        if (is_numeric($query)) {
+            $station = TrainStation::find($query);
```

Comment:
This seems like a bad solution to me. This will completely restrict to a) internal IDs and b) only cached stations.

Maybe we should think about using a flag for numeric values to distinguish between ids and ibnr?